vignettes/Model.Rmd

```diff
@@ -10,90 +12,365 @@ vignette: >
-    E
-    L
-    P
-    label='ODE'
-  }
-
-  node [shape = rectangle]
-  Sm
-  Pm
-  Im
-  NonExistent
+  sapply(3:4, function(x){
+    points(x = Mos_pops[,1], y = Mos_pops[,x], type = "l", col = cols[x])})
   
-  # edge definitions with the node IDs
-  E -> L -> P [color='green', label='1']
-  P -> Sm [color='green', label='2']
-  Sm -> Pm -> Im [color='orange', label='3']
-  Sm -> NonExistent [color='red', label='4']
-  Pm -> NonExistent [color='red', label='4']
-  Im -> NonExistent [color='red', label='4']
-  }")
+  legend("topright", legend = c("A. arab","A. fun","A. gamb"),
+         col = cols[-1], lty = 1, bty = "n", ncol = 1)
+}
+
+species_plot(Mos_sp_dist_sim)
+```
+
+### Seasonality
+
+The `malariasimulation` package has the capacity to simulate malaria transmission for a range of seasonal transmission profiles. This is achieved by specifying an annual rainfall profile that shapes mosquito population dynamics, thereby impacting malaria transmission.
+
+To include seasonality, we must set the parameter `model_seasonality = TRUE` and assign values to parameters that determine seasonality: `g0`, `g` and `h`. These parameters must be set directly by passing a list of named parameters to the `overrides` argument of the `get_parameters()` function.
+
+```{r, fig.align = 'center', fig.height = 5, fig.width = 7}
+# Set parameters, inclusing seasonality parameters
+params_seasons <- get_parameters(overrides = list(
+  model_seasonality = TRUE,
+  g0 = 0.28,
+  g = c(0.2, -0.07, -0.001),
+  h = c(0.2, -0.07, -0.1)))
+
+# Run simulation
+seasonality_simulation <- run_simulation(timesteps = 600, parameters = params_seasons)
+
+# Collect results
+All_mos_cols <- paste0(c("E","L","P","Sm","Pm", "Im", "NonExistent"),"_All_count")
+
+# Plot results
+plot(seasonality_simulation[,1], rowSums(seasonality_simulation[,All_mos_cols]),
+     ylim = c(0, 200000), type = "l", xlab = "Days", ylab = "Mosquito population size")
+grid()
+### Run simulation using seasonality over previous species distribution
+# Update seasonality parameter set to include species distributions
+params_seasons_species <- set_species(parameters = params_seasons,
+                                      species = list(arab_params, fun_params, gamb_params),
+                                      proportions = c(0.1,0.3,0.6))
+
+# Run simulation
+seasonality_species_simulation <- run_simulation(timesteps = 600,
+                                                 parameters = params_seasons_species)
+
+# Collect results
+Mos_season_species_sim <- cbind(seasonality_species_simulation$timestep,
+                         rowSums(seasonality_species_simulation[,arab_cols]),
+                         rowSums(seasonality_species_simulation[,fun_cols]),
+                         rowSums(seasonality_species_simulation[,gamb_cols]))
+
+# Plot results
+species_plot(Mos_season_species_sim)
+
 ```
+
+The mosquito population size is no longer constant and follows the patterns set by rainfall.
```

Comment:
really nice example! 

What is going on with the A. arab population - does it just die out? Might need to add an explanation

Reply:
I'm not sure what you mean - this is the plot I get...

![image](https://user-images.githubusercontent.com/73181123/229086211-4fe6ec7b-aa0c-46bd-b964-2797a11bebf4.png)

A. arab stays at around 100,000